### DISCUSSION TITLE
Help with Linux system calls

### DISCUSSION THREAD

**Discussion by ghost** [community/community#88589] [Programming Help]
### Body

What are all of the Linux system call numbers?

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by faridnec** [comment]
Depends on the Linux kernel version. You can find the system call numbers in the system call table in the Linux kernel source code. The table is typically located in the arch/<your-architecture>/kernel/syscalls/syscall_64.tbl file for 64-bit architectures or arch/<your-architecture>/kernel/syscalls/syscall_32.tbl for 32-bit architectures.

e.g. on an x86-64 architecture, you can find the system call numbers in the arch/x86/entry/syscalls/syscall_64.tbl file.

**Reply by ghost** [reply]
I use Lubuntu and Kali

**Reply by faridnec** [reply]
the system call numbers will be consistent across different distributions running the same version of the Linux kernel. To find the system call numbers, you would need to refer to the system call table in the Linux kernel source code, as mentioned in the previous answer. The location of this table may vary slightly depending on the architecture and kernel version.

For example, on an x86-64 architecture, you can typically find the system call numbers in the arch/x86/entry/syscalls/syscall_64.tbl file in the Linux kernel source code directory.

**Comment by naveen-98** [comment]
Hey @PyDOS8 

Due to the sheer number of Linux system call numbers, providing them all in one place wouldn't be practical or helpful. Additionally, the numbers can vary depending on the specific kernel version and architecture.

However, I can offer a few options to explore Linux system call numbers:

1. Searchable Online Tables:

Filippo.io Linux Syscall Table: https://blog.rchapman.org/posts/Linux_System_Call_Table_for_x86_64/
Searchable Linux Syscall Table for x86 and x86_64: https://github.com/torvalds/linux/blob/master/arch/x86/entry/syscalls/syscall_64.tbl
Chromium OS Docs - Linux System Call Table: https://chromium.googlesource.com/chromiumos/docs/+/HEAD/constants/
These tables allow you to search for specific system calls by name or number, providing details like arguments, calling conventions, and links to further documentation.

2. Kernel Documentation:

The source code for your specific Linux kernel version includes documentation for each system call in the Documentation/syscalls directory. However, this requires familiarity with reading kernel code and understanding technical details.
